### ISSUE TITLE
feat(YouTube): Custom filters whitelist

### ISSUE THREAD

**Issue by Francesco146** [anddea/revanced-patches#397]
### Application

YouTube

### Feature description

Add an option for some custom filters, that are considered to be whitelisted from the current filters.

For example:
• Custom filters:
    `shelf_header`
• Custom filters whitelist:
    `shelf_header$History`

now all `shelf_header` components are hidden, but not the components matching the whitelist filter.

This is useful when a component name is too generic and includes more components in once.

### Motivation

There's plenty of reasons why this could be useful:
* could make some patches easier
* more control over the rendered components 
* more possibilities for hiding stuff
* make possible hiding all components included in a big filter except some of them

### Acknowledgements

- [X] This issue is not a duplicate of an existing feature request.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] I have written the title and contents in English.

**Comment by Francesco146** [comment]
I've posted this so i don't forget, this week I'll take a look if it's a feasible thing for me

**Comment by anddea** [comment]
Good idea, it would be great to collect custom filters (with screenshots), so people know about them. Most of the people don't even know how to use custom filters. 

Okay, I'll wait if you can implement the solution.

**Comment by Francesco146** [comment]
refer to these PR:

integrations: https://github.com/anddea/revanced-integrations/pull/9

patches: https://github.com/anddea/revanced-patches/pull/409
